src/main/java/application/UtilController.java

```diff
@@ -97,6 +108,36 @@ public NotificationMessage clearUserData(
         return notificationMessage;
     }
 
+    @ApiOperation(value = "Reports back on any locks in place for the current user")
+    @RequestMapping(value = Constants.URL_CHECK_LOCKS, method = RequestMethod.POST)
+    public LockReportBean checkLocks(@RequestBody AuthenticatedRequestBean requestBean) throws Exception {
+        String key = LockAspect.getLockKeyForAuthenticatedBean(requestBean, formSessionRepo);
+
+
+        Integer secondsLocked = userLockRegistry.getTimeLocked(key);
+        if(secondsLocked == null) {
+            return new LockReportBean(false, 0);
+        } else{
+            return new LockReportBean(true, secondsLocked);
+        }
+    }
+
+    @ApiOperation(value = "Breaks any currently locked requests for the current user")
+    @RequestMapping(value = Constants.URL_BREAK_LOCKS, method = RequestMethod.POST)
```

Comment:
Is there an user action associated with this on HQ side ? Or HQ is just goign to poll Formplayer periodically to check for any long running locks and subsequently timeout these ?

Reply:
There's a manual action currently in this PR that shows up under the settings menu: 
https://github.com/dimagi/commcare-hq/pull/27430

but not sure what the final presentation will be (like a popup in a new screen trying to do a request that says "Waiting for another action, do you want to cancel it?"), there. My guess is that we'll need to split out accessors to the session by "read only" or something different to allow some long running requests to be made in parallel.